test/src/folding/parsers/highlight_parser_test.dart

```diff
@@ -10,142 +10,126 @@ import 'package:flutter_code_editor/src/single_line_comments/single_line_comment
 import 'package:flutter_test/flutter_test.dart';
 import 'package:highlight/highlight_core.dart';
 import 'package:highlight/languages/java.dart';
+import 'package:highlight/languages/scala.dart';
 
 void main() {
-  group('HighlightFoldableBlockParser', () {
-    test('parses', () {
-      final examples = [
-        //
-        _Example(
-          'Java. Empty string',
-          code: '',
-          mode: java,
-          expected: const [],
-          expectedInvalid: const [],
-        ),
-
-        _Example(
-          'Java. Multiline pair-character blocks',
-          code: '''
-public class MyClass {
-  public static void main(
-    String[
-    ] args) {
-    }
+  _testJava();
+  _testScala();
 }
-''',
-          mode: java,
-          expected: const [
-            _FB(startLine: 0, endLine: 5, type: _T.braces),
-            _FB(startLine: 1, endLine: 3, type: _T.parentheses),
-            _FB(startLine: 2, endLine: 3, type: _T.brackets),
-            _FB(startLine: 3, endLine: 4, type: _T.braces),
-          ],
-          expectedInvalid: const [],
-        ),
-
-        _Example(
-          'Java. Single-line pair-character blocks are ignored',
-          code: '''
+
+void _testJava() {
+  group('HighlightFoldableBlockParser for Java.', () {
+    test('Java. Empty string.', () {
+      _parseAndCheck(mode: java, code: '', expected: []);
+    });
+
+    test('Java. Multiline pair-character blocks', () {
+      const code = '''
 public class MyClass {
-  public static void main(String[] args) {}
-}
-''',
-          mode: java,
-          expected: const [
-            _FB(startLine: 0, endLine: 2, type: _T.braces),
-          ],
-          expectedInvalid: const [],
-        ),
-
-        _Example(
-          'Java. Blocks with missing pair characters',
-          code: '''
+public static void main(
+  String[
+  ] args) {
+  }
+}''';
+      const expectedBlocks = [
+        _FB(startLine: 0, endLine: 5, type: _T.braces),
+        _FB(startLine: 1, endLine: 3, type: _T.parentheses),
+        _FB(startLine: 2, endLine: 3, type: _T.brackets),
+        _FB(startLine: 3, endLine: 4, type: _T.braces),
+      ];
+      _parseAndCheck(mode: java, code: code, expected: expectedBlocks);
+    });
+
+    test('Java. Single-line pair-character blocks are ignored.', () {
```

Comment:
No need to repeat the language here because 'Java' is already in the group name.
https://medium.com/flutter-senior/how-to-name-and-group-tests-in-a-dart-file-553e194bd8af